rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -938,14 +938,19 @@ private void processSigning(BtcECKey federatorPublicKey, List<byte[]> signatures
             }
         }
 
+
+        int missingSignatures = BridgeUtils.countMissingSignatures(btcContext, btcTx);
+
         // If tx fully signed
-        if (hasEnoughSignatures(btcTx)) {
+        if (missingSignatures == 0) {
             logger.info("Tx fully signed {}. Hex: {}", btcTx, Hex.toHexString(btcTx.bitcoinSerialize()));
             provider.getRskTxsWaitingForSignatures().remove(new Keccak256(rskTxHash));
 
             eventLogger.logReleaseBtc(btcTx, rskTxHash);
         } else {
-            logger.debug("Tx not yet fully signed {}.", new Keccak256(rskTxHash));
+            int neededSignatures = (getFederationSize() / 2) + 1;
```

Comment:
You should obtain the size of the affected Federation, not of the active (which is what `getFederationSize` does).
Besides this, instead of calculating the requirement like this, use the existing method from Federation `federation.getNumberOfSignaturesRequired()`